src/components/ProductCardContainer/ProductCard.jsx

```diff
@@ -0,0 +1,91 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  namePriceContainer,
+  productCardStyles,
+  productImageStyles,
+  productInfoStyles,
+  addButtonStyles,
+  removeButtonStyles,
+  productName,
+  productDescribtion,
+  productPrice,
+} from './styles';
+import Button from '../ButtonComponent/Button';
+
+const ProductCard = ({ product, addToCart, isInCart, onRemove }) => {
+  const [quantity, setQuantity] = useState(1);
+
+  const handleAddToCart = () => {
+    addToCart({ ...product, quantity });
+  };
+
+  const handleQuantityChange = (event) => {
+    setQuantity(Number(event.target.value));
+  };
+
+  return (
+    <div style={productCardStyles}>
+      <div style={productImageStyles}>
+        <img src={product.img} alt={product.meal} style={{ width: 120, height: 120 }} />
+      </div>
+
+      <div style={productInfoStyles}>
+        <div style={namePriceContainer}>
+          <h2 style={productName}>{product.meal}</h2>
+          <p style={productPrice}>
+            $ {isInCart ? (product.price * product.quantity).toFixed(2) : (product.price * quantity).toFixed(2)} USD
```

Comment:
All these calculations and checks better to do in a separate constant, but not in jsx